src/redux/reducers/main.ts

```diff
@@ -265,43 +271,66 @@ export const mainSlice = createSlice({
     /**
      * Updates the content of the specified resource to the specified value
      */
-    updateResource: {
-      reducer: (state: Draft<AppState>, action: PayloadAction<UpdateResourcePayload>) => {
-        try {
-          const resource = state.resourceMap[action.payload.resourceId];
+    updateResource: (state: Draft<AppState>, action: PayloadAction<UpdateResourcePayload>) => {
+      try {
+        const resource = state.resourceMap[action.payload.resourceId];
+        if (resource) {
+          if (isFileResource(resource)) {
+            const updatedFileText = saveResource(resource, action.payload.content, state.fileMap);
+            resource.text = updatedFileText;
+            resource.content = parseDocument(updatedFileText).toJS();
+            recalculateResourceRanges(resource, state);
+          } else {
+            resource.text = action.payload.content;
+            resource.content = parseDocument(action.payload.content).toJS();
+          }
+
+          let resourceIds = findResourcesToReprocess(resource, state.resourceMap);
```

Comment:
this could be extracted to a method used here and below!?

Reply:
sure, will do